builtin/credential/aws/path_login.go

```diff
@@ -801,11 +801,18 @@ func (b *backend) pathLoginUpdateEc2(ctx context.Context, req *logical.Request,
 	}
 
 	if roleEntry.MaxTTL > time.Duration(0) {
+		shortestTTL := b.System().DefaultLeaseTTL()
+		if roleEntry.TTL > time.Duration(0) && roleEntry.TTL < shortestTTL {
+			shortestTTL = roleEntry.TTL
```

Comment:
We don't actually want to do this.

Role entry TTLs override the system default TTL so long as they are less than the given max TTL. And the max TTL is enforced by core.